code/datums/repositories/decls.dm

```diff
@@ -43,11 +126,30 @@
 		. = get_decls(subtypesof(decl_prototype))
 		fetched_decl_subtypes[decl_prototype] = .
 
+/decl
+	abstract_type = /decl
+	var/uid
+	var/decl_flags = null // DECL_FLAG_ALLOW_ABSTRACT_INIT, DECL_FLAG_MANDATORY_UID
+	var/initialized = FALSE
+
 /decl/proc/Initialize()
 	SHOULD_CALL_PARENT(TRUE)
 	SHOULD_NOT_SLEEP(TRUE)
+	if(initialized)
+		CRASH("[type] initialized more than once!")
+	initialized = TRUE
+	return INITIALIZE_HINT_NORMAL
+
+/decl/proc/validate()
```

Comment:
this is useless because you didn't port the unit test that actually checks decl validation